packages/driver/cypress/integration/commands/assertions_spec.js

```diff
@@ -2927,4 +2927,23 @@ describe('src/cy/commands/assertions', () => {
       cy.get('.foo').should('not.exist')
     })
   })
+
+  context('implicit assertions', () => {
+    // https://github.com/cypress-io/cypress/issues/18549
+    it('only snapshots once when failing to find DOM elements', (done) => {
+      sinon.spy(cy, 'createSnapshot')
+      cy.on('fail', (err) => {
+        // We have a regression where snapshots are being taken for every failed retry on implicit assertions
+        // for element existence. The exact number expected is unclear (and this test should be updated with a
+        // more exact assertion once we've solved the regression) - but it definitely shouldn't be above 5!
+        expect(cy.createSnapshot.callCount).to.be.lessThan(5)
+
+        cy.createSnapshot.restore()
```

Comment:
you don't need to do this, cypress automatically cleans up all mocks/stubs in between tests, so its impossible to leak. you only need to restore if you need to restore and then mock it again for some reason within *the same test*, which is virtually and effectively "never"